homeassistant/components/smarttub/climate.py

```diff
@@ -109,3 +130,9 @@ async def async_set_temperature(self, **kwargs):
         temperature = kwargs[ATTR_TEMPERATURE]
         await self.spa.set_temperature(temperature)
         await self.coordinator.async_refresh()
+
+    async def async_set_preset_mode(self, preset_mode: str):
+        """Activate the specified preset mode."""
+        heat_mode = next(k for k, v in self.PRESET_MODES.items() if v == preset_mode)
```

Comment:
Crease a reverse dict instead of doing a linear search

Reply:
OK. I thought it would be fine since the dicts have only 2-3 items but I'm happy to add it

Reply:
It won't make much difference at 3 items but there type of data structures have a tendency to get larger over time